### ISSUE TITLE
If ara-api cannot read settings.yml, it falls back to some default w/o warning

### ISSUE THREAD

**Issue by zswanson** [ansible-community/ara#117]
## What component is this about ?

ara-api

## What is your ARA installation like ?

ara-api was installed on an ubuntu AMI. I was attempting to install it to /opt/ara and *not* use the `.ara` hidden path, and use a service user with home directory at /opt/ara.

## What is happening ?

My install had mistakenly set the ownership of the ara/server/settings.yaml to root, with no read permissions for my ara service user. When ara-api launched, it appears to have fallen back to an internal default settings, while *still reporting that it was using the config file I had specified*. It took a few days of debugging and reinstalling before I realized the error.

## What should be happening ?
I think that falling back to the internal defaults is fine but some sort of error or warning needs to be provided.

**Comment by dmsimard** [comment]
Hi @zswanson, thanks for creating the issue and apologies for the delay.

I've reproduced this issue by *not* having a ``~/.ara`` directory and doing ``export ARA_SETTINGS=/root/foo.yml`` (which doesn't exist and even if it did, the user doesn't have privilege to /root):
```
[ara] No setting found for SECRET_KEY. Generating a random key...
[ara] Creating data & configuration directory: /home/dmsimard/.ara/server
[ara] No setting found for SECRET_KEY. Generating a random key...
2020-04-13 15:11:36,995 INFO django.utils.autoreload: Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).

You have 23 unapplied migration(s). Your project may not work properly until you apply the migrations for app(s): admin, api, auth, contenttypes, db, sessions.
Run 'python manage.py migrate' to apply them.

April 13, 2020 - 15:11:37
Django version 2.2.9, using settings 'ara.server.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
^C[ara] Using settings file: /root/foo.yml
[ara] Using settings file: /root/foo.yml
```

Out of curiosity, I tried doing ``export ARA_BASE_DIR=/root/foo`` instead and got the error I expected:
```
[ara] No setting found for SECRET_KEY. Generating a random key...
[ara] Creating data & configuration directory: /root/foo
Traceback (most recent call last):
  File ".tox/ansible-integration/bin/ara-manage", line 10, in <module>
    sys.exit(main())
  File "/home/dmsimard/dev/git/ansible-community/ara/ara/server/__main__.py", line 35, in main
    if settings.DATABASE_ENGINE == "django.db.backends.postgresql":
  File "/home/dmsimard/dev/git/ansible-community/ara/.tox/ansible-integration/lib/python3.7/site-packages/django/conf/__init__.py", line 79, in __getattr__
    self._setup(name)
  File "/home/dmsimard/dev/git/ansible-community/ara/.tox/ansible-integration/lib/python3.7/site-packages/django/conf/__init__.py", line 66, in _setup
    self._wrapped = Settings(settings_module)
  File "/home/dmsimard/dev/git/ansible-community/ara/.tox/ansible-integration/lib/python3.7/site-packages/django/conf/__init__.py", line 157, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/home/dmsimard/dev/git/ansible-community/ara/.tox/ansible-integration/lib64/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dmsimard/dev/git/ansible-community/ara/ara/server/settings.py", line 243, in <module>
    os.makedirs(BASE_DIR, mode=0o700)
  File "/home/dmsimard/dev/git/ansible-community/ara/.tox/ansible-integration/lib64/python3.7/os.py", line 221, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/root/foo'
```

So it seems the issue is only occurring when using ``ARA_SETTINGS`` by itself. Could you confirm if that was also your experience ?

This is fixable but I want to make sure we're both talking about the same thing.

**Comment by zswanson** [comment]
That's correct, I was running ARA by systemd unit and ONLY setting the ARA_SETTINGS value.

**Comment by dmsimard** [comment]
Thanks! I've submitted a patch to address the issue: https://review.opendev.org/#/c/720072/
I'll close the issue once the change merges.

It will be included in the upcoming release of 1.4 which I am hoping to wrap up soon.
